### ISSUE TITLE
private key auth does not work (no meaningful error)

### ISSUE THREAD

**Issue by flying-sheep** [WebFreak001/code-debug#95]
- [x] `gdb --version` >= 7.7.1
- [x] it works on the command line with `gdb`
- [x] `cwd` and `target` are properly set

password works, but as you said, it’s a terrible idea. trying to use a private key merely gives:

```
Could not run gdb over ssh!
Error: All configured authentication methods failed
```

running `ssh <user>@<target>` also works of course

**Comment by WebFreak001** [comment]
Are you sure you used the correct key file? Move all other key files (especially in your .ssh folder) temporarily to some other place. About the error message: the ssh library I use for connecting ([ssh2](https://github.com/mscdex/ssh2)) doesn't really give any more information. So no `-v`/`-vvv` options there.

For the original issue: SSH using a keyfile works for me, check again if you used the correct keyfile, it won't automatically use the ones in your `.ssh` folder

**Comment by flying-sheep** [comment]
yes, my key works. i did pretty much this, since my key had a passphrase your plugin/library couldn’t handle:

```console
$ ssh $user@$server 'rm .ssh/authorized_keys'
$ rm ~/.ssh/id_rsa{,.pub}
$ ls ~/.ssh
authorized_keys  config  known_hosts
$ ssh-keygen -t rsa
Generating public/private rsa key pair.
Enter file in which to save the key ($HOME/.ssh/id_rsa): 
Enter passphrase (empty for no passphrase): 
Enter same passphrase again: 
Your identification has been saved in $HOME/.ssh/id_rsa.
Your public key has been saved in $HOME/.ssh/id_rsa.pub.
[…]
$ ssh-copy-id $user@$server
$ ssh $user@$server 'echo shit works'
shit works
```

**Comment by WebFreak001** [comment]
Hm it might be because of a passphrase on the key file?

**Comment by flying-sheep** [comment]
no: i did the above because my *previous* key had one, and your plugin’s output said that it couldn’t handle that. so i deleted my old key and created a new one without passphrase, which then proceeded to also not work.

that’s the one i filed this issue for.

**Comment by GitMensch** [comment]
Same for me with 0.25.1 and (with updated ssh module of my PR's 0.26-dev) - password work, key works via Terminal + SSH but not via this extension.

**Comment by Camille019** [comment]
Same for me with 0.27.0
